src/createClassProxy.js

```diff
@@ -44,17 +44,56 @@ function proxyClass(InitialComponent) {
     return !isEqualDescriptor(staticDescriptors[key], currentDescriptor);
   }
 
+  function instantiate(factory, context, params) {
+    const component = factory();
+
+    try {
+      return component.apply(context, params);
+    } catch (err) {
+      // Native ES6 class instantiation
+      let instance;
+
+      switch (params.length) {
```

Comment:
Yes, I think the issue here is with wrong test methodology.
I think we need to change the code to Object.define function name. 
If it works we can remove both dynamic evaluation and this test.